### ISSUE TITLE
Allow Aurelia to instantiate Custom Elements and Attached Behaviors in the existing body (for MPA)

### ISSUE THREAD

**Issue by Aaike** [aurelia/framework#58]
This is for the use case where a page already exists and has content in the body and we want Aurelia to simply initialize the custom elements and attached behaviors on that page.

A new attribute could be added to the bootstrapper, something like `aurelia-app-inline` which will cause Aurelia to parse the content in the body rather than using the router to load in page content.

This will likely cause FOUC but this can be prevented by adding a class to the body which will be removed by Aurelia once everything is ready.

The body class could be optional and instead a class can be added to each individual element where FOUC is expected. The class is then also removed by aurelia once the component is created.
This could be done in an attempt to solve the FOUC issue on a per element basis

What would be even better is that the `::unresolved` selector for HTML custom elements would work with Aurelia components (similar to http://www.html5rocks.com/en/tutorials/webcomponents/customelements/#gettingstarted)
or maybe https://www.polymer-project.org/docs/polymer/styling.html

But i suppose a simple class is just fine.

**Comment by Aaike** [comment]
Another case is where markdown content is rendered in the backend and it produces the markup for me, so It does not have any special attributes , only a language-*** class

Maybe i will need a hook to add attributes to certain elements before Aurelia starts the activation process of the components on the page.

**Comment by Mitranim** [comment]
Having the exact same issue / use case. I have a website that is mostly static, but is progressively enhanced with some dynamic components that use xhr/websocket for server interaction. I need to instantiate the aurelia app on the document body without replacing the existing content, but activating any custom elements found. Would be nice to have a blessed way to cover this use case.

**Comment by devmondo** [comment]
:+1: i have same issue as guys above

**Comment by Aaike** [comment]
this is a reference to the [conversation that started this issue](https://gitter.im/Aurelia/Discuss?at=54f0ba792095620968ab6c09)

This issue was actually because @EisenbergEffect requested it as seen in that chat, i guess it's just not high on the priority list yet ;)

**Comment by EisenbergEffect** [comment]
We needed to address a few other core issues related to the templating engine first. This is issue important. It's also related to several other similar issues such as dynamic HTML content with custom elements and using Aurelia elements outside of an Aurelia app. These three features are likely to share a bunch of code, but some of it is fundamentally different than existing implementations for the templating engine. So, it's not a small effort and we'd like to see if we can addresses multiple issues as part of the solution. Hang in there...

**Comment by zewa666** [comment]
I guess we can close this old one @EisenbergEffect since this should now be handled by aurelias new progressive enhancement feature right?

**Comment by EisenbergEffect** [comment]
It is now possible, technically, so I will close this. There is another issue tracking a new custom attribute that would make this easier fora number of scenarios.

**Comment by davidpeden3** [comment]
@EisenbergEffect @zewa666 can you guys please link the issue to cross-reference from here so followers of this issue can subscribe to the other issue?

Thanks!

**Comment by EisenbergEffect** [comment]
Here you go: https://github.com/aurelia/templating-resources/issues/38
